staging/src/k8s.io/apimachinery/pkg/util/errors/errors.go

```diff
@@ -147,6 +147,28 @@ func Flatten(agg Aggregate) Aggregate {
 	return NewAggregate(result)
 }
 
+// UniqueAggregateWithCount removes duplicates from flattened Aggregate
+// and returns count for each unique error present in the Aggregate
+func UniqueAggregateWithCount(agg Aggregate) Aggregate {
+	if agg == nil || len(agg.Errors()) == 0 {
+		return agg
+	}
+	errs := Flatten(agg).Errors()
```

Comment:
The point is you can bin on insertion and prevent the need to re-run through it, the append is only a couple of lines above -  

https://github.com/kubernetes/kubernetes/pull/47003/files#diff-a9420b7e5a53a445d54a716d341874abR192 

If you bin there, then you could have a NewAggregate overload that passes in the error map and you don't need a 2N+Y it's just 2Y.